ReactNativeClient/lib/services/plugins/utils/contentScriptsToCodeMirror.ts

```diff
@@ -0,0 +1,28 @@
+import { PluginStates } from '../reducer';
+import { ExtraRendererRule } from 'lib/joplin-renderer/MdToHtml';
+import { ContentScriptType } from '../api/types';
+
+export default function contentScriptsToCodeMirror(plugins:PluginStates):ExtraRendererRule[] {
+	const output:ExtraRendererRule[] = [];
+
+	for (const pluginId in plugins) {
+		const plugin = plugins[pluginId];
+		const contentScripts = plugin.contentScripts[ContentScriptType.CodeMirrorPlugin];
+		if (!contentScripts) continue;
+
+		for (const contentScript of contentScripts) {
+			const module = require(contentScript.path);
+			if (!module.default || typeof module.default !== 'function') throw new Error(`Content script must export a function under the "default" key: Plugin: ${pluginId}: Script: ${contentScript.id}`);
+
+			const loadedModule = module.default({});
+			if (!loadedModule.plugin) throw new Error(`Content script must export a "plugin" key: Plugin: ${pluginId}: Script: ${contentScript.id}`);
+
+			output.push({
+				id: contentScript.id,
+				module: loadedModule,
+			});
+		}
+	}
+
+	return output;
+}
```

Comment:
It's good that the logic is pretty much the same as for loading markdown-it plugins. But in that case, maybe we should refactor the existing contentScriptsToRendererRules so that it's more generic and use it for both CodeMirror and MarkdownIt? From the look of it, it will only need to take an extra "ContentScriptType" parameter.

Reply:
That's what I'm thinking I'll do, I started like this in case I decided to make drastic changes, this was just a good starting point. I don't imagine I will make any sweeping changes because keeping the interface the same between content scripts is pretty valuable.